src/imgix-api.js

```diff
@@ -71,16 +70,11 @@
                 return response.json();
             }
         })
-        .catch(error => {
-            if (response) {
-                throw new APIError(`Request failed with status ${ response.status }.`, error, response.status);
-            } else {
-                throw new APIError(error.toString(), null, 'REQUEST_FAILED');
-            }
-        });
+        .catch(response => Promise.reject(catchError(response)));
     };
 
     const constructUrl = (path, version) => `${ API_URL }/v${ version }/${ path }`;
+    const catchError = (error) => `Request failed with status ${ error.toString() }.`;
```

Comment:
This function should still return a rejected Promise, methinks. Otherwise the user calls `imgixAPIClient.request(...)` and gets back a resolved promise with a string as the value... that looks like a success to me, and it should look like a failure.